roles/tftp-server/templates/preseed.cfg.j2

```diff
@@ -109,7 +109,7 @@ d-i mirror/http/directory string /debian
 d-i mirror/http/hostname string {{ mirror | default('archive.ubuntu.com') }}
 d-i mirror/http/directory string /ubuntu
 {% endif %}
-d-i mirror/http/proxy string {{ apt_proxy }}
+d-i mirror/http/proxy string{% if apt_proxy %}http://{{ inventory_hostname }}:8000/{% endif %}
```

Comment:
correct.
I don't think you can reference inventory_hostname in the inventory/group_vars/all.yml